test/4.public.UrlChecker.js

```diff
@@ -199,5 +199,32 @@ describe("PUBLIC -- UrlChecker", function()
 			instance.enqueue( conn.absoluteUrl+"/fixtures/link-real.html", null, {index:1} );
 			instance.enqueue( conn.absoluteUrl+"/fixtures/link-fake.html", null, {index:2} );
 		});
+
+		it("should check unique url only once", function(done)
+		{
+			var results = [];
+			
+			var options = utils.options();
+			options.checkUniqueUrlOnce = true;
+
+			var instance = new UrlChecker( options,
+			{
+				link: function(result, customData)
+				{
+					results.push(result);
+				},
+				end: function()
+				{
+					expect(results).to.have.length(2);
```

Comment:
Shouldn't it return the cached result? If a broken link exists in multiple places, all should be reported.